ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.config;
+
+import com.github.ambry.utils.Utils;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+/**
+ * Configuration parameters required by a CompressionService.
+ * Configurations are read-only.  The only way to alter the values is by modifying the config files.
+ * The configuration applies to compression only.  Decompression is automatic and has no configuration.
```

Comment:
if there is no compression, will there be decompression ?

Reply:
I mean shouldn't the same config control both compression and decompression ?

Reply:
Compression will store the algorithm name inside the binary, so that decompression can lookup the algorithm by this name.  That's why the compression config is different from decompression.  There aren't that many decompression config.